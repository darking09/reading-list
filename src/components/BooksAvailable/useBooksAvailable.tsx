import { useEffect, useState } from "react";

const singularBaseText = " Libro disponible: ";
const pluralBaseText = " Libros disponibles: ";

export function useBooksAvailable(numberOfBooks : number) {
  const [textToReturn, setTextToReturn] = useState("");

  useEffect(() => {
    if (numberOfBooks === 1) {
      setTextToReturn(`${numberOfBooks} ${singularBaseText}`)
      return;
    }
    setTextToReturn(`${numberOfBooks} ${pluralBaseText}`);
  }, [numberOfBooks]);

  return {
    textToReturn
  };
}
