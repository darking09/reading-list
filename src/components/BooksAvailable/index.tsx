'use client'
import { BooksAvailableProps } from '@typesFiles/props/booksAvailable'
import { useBooksAvailable } from './useBooksAvailable'

export default function BooksAvailable ({ numberOfBooks } : BooksAvailableProps) {
  const { textToReturn } = useBooksAvailable(numberOfBooks);

  return (
    <div className="flex flex-row p-2">
      {textToReturn}
    </div>
  )
}
