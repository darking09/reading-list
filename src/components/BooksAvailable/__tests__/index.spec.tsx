import BooksAvailable from '../index'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'

describe('BooksAvailable', () => {
  it('should render', () => {
    const { container } = render(
      <BooksAvailable
        numberOfBooks={{
          numberOfAvailableBooks: 1,
          numberOfBooksInReadingList: 1,
        }}
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render a book with a cover', () => {
    render(
      <BooksAvailable
        numberOfBooks={{
          numberOfAvailableBooks: 1,
          numberOfBooksInReadingList: 1,
        }}
      />
    )

    const element = screen.getByText(/1 Libro disponible:/)
    expect(element).toBeInTheDocument()
  })
})
