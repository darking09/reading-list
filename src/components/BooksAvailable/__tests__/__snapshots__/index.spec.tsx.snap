// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BooksAvailable should render 1`] = `
<div>
  <div
    class="grid grid-rows-2 pt-2"
  >
    <h1
      class="row-span-1 text-4xl"
    >
      1  Libro disponible: 
    </h1>
    <h2
      class="text-2xl"
    >
      1  Libro en la lista de lectura
    </h2>
  </div>
</div>
`;
